crates/interledger-settlement-engines/src/engines/ethereum_ledger/eth_engine.rs

```diff
@@ -259,160 +231,267 @@ where
 
     /// Routine for notifying the connector about incoming transactions.
     /// Algorithm (heavily unoptimized):
-    /// 1. Fetch the last observed block number and account balance from the store
+    /// 1. Fetch the last observed block number
     /// 2. Fetch the current block number from Ethereum
-    /// 3. Check the current balance of the account
-    /// 4. If the balance is not larger than the last observed balance, terminate.
-    /// 5. Fetch all blocks since the last observed one,
+    /// 3. Fetch all blocks since the last observed one,
     ///    until $(current block number - confirmations), where $confirmations
     ///    is a security parameter to be safe against block reorgs.
-    /// 6. For each block (in parallel):
+    /// 4. For each block (in parallel):
     ///     For each transaction (in parallel):
     ///     1. Skip if it is not sent to our address or have 0 value.
     ///     2. Fetch the id that matches its sender from the store
     ///     3. Notify the connector about it by making a POST request to the connector's
     ///        /accounts/$id/settlement endpoint with transaction's value as the
     ///        body. This call is retried if it fails.
-    /// 7. Save the (current block number - confirmations) and current account
-    ///    balance to the store, to be used as last observed data for the next
-    ///    call of this function.
+    /// 5. Save (current block number - confirmations) to be used as
+    ///    last observed data for the next call of this function.
+    // Optimization: This will make 1 settlement API request to
+    // the connector per transaction that comes to us. Since
+    // we're scanning across multiple blocks, we want to
+    // optimize it so that: it gathers a mapping of
+    // (accountId=>amount) across all transactions and blocks, and
+    // only makes 1 transaction per accountId with the
+    // appropriate amount.
     pub fn notify(&self) -> impl Future<Item = (), Error = ()> + Send {
```

Comment:
How about:
```suggestion
     pub fn handle_received_transactions(&self) -> impl Future<Item = (), Error = ()> + Send {
```

This is a minor point but "notify" makes me wonder what it's notifying about